### PR TITLE
Consider root_dir and cwd in ssh_sftpd, if both are provided

### PR DESCRIPTION
**Pull request by kape1395**
The SFTPD server should use root_dir and cwd when resolving
file paths, if both are provided. The root directory should
be used for resolving absolute file names, and cwd should
be used for resolving relative paths.

### CODE REVIEW

File: lib/ssh/src/ssh_sftpd.erl [hunk 1]
```diff
@@ -742,26 +742,29 @@ resolve_symlinks_2([], State, _LinkCnt, AccPath) ->
...
+	false -> Root
+    end,
+    AbsFile = case make_relative_filename(File) of
+	File    -> relate_filename_to_path(File,    CWD1, Canonicalize);
+	RelFile -> relate_filename_to_path(RelFile, Root, Canonicalize)
+    end,
+    case is_within_root(Root, AbsFile) of
+	true  -> AbsFile;
+	false -> Root
     end.
```
Comment (HansN) [inline lib/ssh/src/ssh_sftpd.erl]: The three case statements above has a bit unusual indentation for being Erlang.  Please follow the usual conventions to make it  easier to read ('e' in 'end'  on same column as 'c' in the corresponding 'case' etc).

Comment (kape1395) [inline lib/ssh/src/ssh_sftpd.erl]: I fixed the code indentation, rebased on new maint and squashed the commits.

**Comment by HansN** [comment]
Thanks for the updates, I've added the branch to our nightly tests.

**Comment by HansN** [comment]
Merged!  Thanks for the PR.

**Pull request merged**